src/crypto/crypto_keys.cc

```diff
@@ -1134,6 +1136,51 @@ void KeyObjectHandle::InitEDRaw(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(true);
 }
 
+void KeyObjectHandle::Equals(const FunctionCallbackInfo<Value>& args) {
+  std::shared_ptr<KeyObjectData> key =
+    Unwrap<KeyObjectHandle>(args.Holder())->Data();
+  std::shared_ptr<KeyObjectData> key2 =
+    Unwrap<KeyObjectHandle>(args[0].As<Object>())->Data();
+
+  KeyType keyType = key->GetKeyType();
+  CHECK_EQ(keyType, key2->GetKeyType());
+
+  bool ret;
+  switch (keyType) {
+    case kKeyTypeSecret: {
+      size_t size = key->GetSymmetricKeySize();
+      if (size == key2->GetSymmetricKeySize())
+        ret = CRYPTO_memcmp(
+          key->GetSymmetricKey(),
+          key2->GetSymmetricKey(),
+          size) == 0;
+      else
+        ret = false;
+      break;
+    }
+    case kKeyTypePublic:
+      // Fall through
```

Comment:
This comment is optional when the `case` is empty, but feel free to keep it here.